### PR TITLE
feat: pass api key from mystique to backend

### PR DESCRIPTION
**Pull request by ilhaoni**

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -144,7 +144,9 @@ custom:
       - ${self:provider.environment.TARGET_DATA_BUCKET}
   prune:
     automatic: true
-    number: 5
+    number: 3
```
Comment (sebastianquek) [inline serverless.yml:147]: What's this change ah?

Comment (ilhaoni) [reply serverless.yml:147]: to reduce the number of versions we keep, to reduce our overall lambda storage :x 

sorry i squeezed this in, should i remove ah?

File: src/functionHandlers/proxy/proxy.ts [hunk 2]
```diff
@@ -24,6 +24,9 @@ const handleRequest = async (event: APIGatewayEvent) => {
   }
   const idMapperWithValidator = withAuthValidator(idMapper, isAuthenticated);
```
Comment (sebastianquek) [inline src/functionHandlers/proxy/proxy.ts:25]: `isAuthenticated` will be false here, is this a problem?

Comment (ilhaoni) [reply src/functionHandlers/proxy/proxy.ts:25]: so far no sia

Comment (ilhaoni) [reply src/functionHandlers/proxy/proxy.ts:25]: i tested and it works :O

Comment (sebastianquek) [reply src/functionHandlers/proxy/proxy.ts:25]: oh no LOL, ok if it works let's leave it first, we can revisit and add more tests for this again

**Review by sebastianquek — APPROVED** [review]

File: src/functionHandlers/proxy/proxy.ts (hunk 2 above)
Comment (ilhaoni) [reply src/functionHandlers/proxy/proxy.ts:25]: okk~~ sounds good haha :D

**Pull request merged**